packages/react/src/components/Button/Button.js

```diff
@@ -63,7 +66,9 @@ const Button = React.forwardRef(function Button(
     ref,
   };
 
-  const buttonImage = !ButtonImageElement ? null : (
+  const buttonImage = performingAction ? (
+    <Loading small withOverlay={false} />
```

Comment:
Given the concern raised in the team, would you just put `<Loading>` in the story instead of backing it into `<Button>`?

Reply:
The prevalence of this pattern is great enough that our org is supporting this enhancement in our addons library so if the team does not feel this is an appropriate addition we can continue to support our own button.  But, since the UX is not unique to IoT I thought it may be a good idea to contribute back to the Carbon library so that it may be implemented consistently for all teams in need of this pattern. The button would still function the same for those who do not need a "performing action" state and the implementation in Carbon components would allow us to remove one more deviation from the core library. 

If there are improvements that are not being considered I would love to add those to this PR but I think the UX need is great enough to bake this enhancement into the component itself for the sake of consistency.

Reply:
@davidicus Thank you for sharing your thoughts. Our component library has "Write programs that do one thing and do it well. Write programs to work together" principle, and the approach of putting `<Loading>` just as button child (instead of baking spinner into our `<Button>`) looks aligning better with such principle. If you see some specific difficulty with that feel free to bring up, that may help us system-design our components in future.